service/src/main/java/bio/terra/externalcreds/config/ProviderPropertiesInterface.java

```diff
@@ -28,6 +28,8 @@ public interface ProviderPropertiesInterface {
 
   String getExternalIdClaim();
 
+  String getUserNameAttributeName();
```

Comment:
what is this used for?

Reply:
This gets used when calling the POST oauthcode endpoint (I tested the authorization-url and account linking together to see what I needed to include in the github ClientRegistration). After the token exchange is successful, ECM calls the provider's user info endpoint and [this call](https://github.com/DataBiosphere/terra-external-credentials-manager/blob/dev/service/src/main/java/bio/terra/externalcreds/services/OAuth2Service.java#L137) was failing because that attribute wasn't defined.

Reply:
Also, the field I chose to be the `userNameAttributeName` is "login" which is the github username. This will be stored as the user's `externalUserId` in the linked account table.

Reply:
RAS has a lot of these values set by default (here is its `userNameAttributeName`): https://github.com/spring-projects/spring-security/blob/6.1.6/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistrations.java#L254